### PR TITLE
fixup: Trip planner itinerary tagging

### PR DESCRIPTION
**Pull request by thecristen**
#### Summary of changes

**Asana Ticket:** [Trip planner itinerary tagging: implement tie-breaking & new "most direct" tag (tie-breaking with fallback)](https://app.asana.com/0/555089885850811/1206595270365336/f)

Sorry, there's not much here! Most of the actual implementation of the new tag, as well as implementation of tagging in general, is in https://github.com/thecristen/open_trip_planner_client/compare/v0.5.0...v0.6.1 in the `OpenTripPlannerClient.ItineraryTag` and `OpenTripPlannerClient.ItineraryTag.MostDirect` modules.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -109,12 +111,13 @@ defmodule DotCom.Mixfile do
       {:logster, "1.1.1"},
       {:mail, "0.3.1"},
       {:mock, "0.3.8", [only: :test]},
+      {:mox, "1.1.0", [only: :test]},
```
Comment (kotva006) [inline mix.exs:114]: These seem to be added as dependencies but not used in this pull request

**Comment by kotva006** [comment]
Is the goal of the [open_trip_planner_client](https://github.com/thecristen/open_trip_planner_client/) to replace the implementation of the `open_trip_planner.ex`?
I am confused because they both seem to still exist and be doing similar things

File: mix.exs (hunk 1 above)
Comment (thecristen) [reply mix.exs:114]: The commit message that added these was `[fixup merge commit](https://github.com/mbta/dotcom/pull/1897/commits/3313b4834a0fe3dc655a30dacc793d3793a98501)`....these were supposed to be in the `feat/trip-planner-improvements` branch (and are being used there already).

**Comment by thecristen** [comment]
@kotva006 Yeah, just the `plan/3` function specifically. See the description of a previous PR to this branch for more: https://github.com/mbta/dotcom/pull/1883

**Comment by kotva006** [comment]
> @kotva006 Yeah, just the `plan/3` function specifically. See the description of a previous PR to this branch for more: #1883

I am not tracking.  If the `open_trip_planner_client` library replaces the `plan/3` function, why does the function still exist?

**Comment by thecristen** [comment]
@kotva006 Did you see the source? https://github.com/mbta/dotcom/blob/cbj/fix-tags/lib/trip_plan/api/open_trip_planner.ex#L18 

Our `plan/3` function calls the function returned by `Application.get_env(:dotcom, :trip_planner, OpenTripPlannerClient)`. In prod/dev that just refers to the `OpenTripPlannerClient.plan/3` function, but in `MIX_ENV=test` it's configured to point to a different function that's used for testing.

**Review by kotva006 — APPROVED** [review]

**Pull request merged**